### PR TITLE
#989 Correct error line and column number

### PR DESCRIPTION
**Pull request by joan38**

### CODE REVIEW

File: bsp/src/mill/bsp/BspLoggedReporter.scala [hunk 1]
```diff
@@ -91,24 +91,25 @@ class BspLoggedReporter(client: bsp.BuildClient,
...
   private[this] def getSingleDiagnostic(problem: Problem): Diagnostic = {
     val pos = problem.position
-    val i: Integer = pos.startLine.orElse(pos.line).getOrElse[Int](0)
-    println(i)
+    val line = pos.line.map(_ - 1) // Zinc's range starts at 1 whereas BSP 0
     val start = new bsp.Position(
-      pos.startLine.orElse(pos.line).getOrElse[Int](0),
-      pos.startOffset.orElse(pos.offset).getOrElse[Int](0)
+      pos.startLine.orElse(line).getOrElse[Int](0),
+      pos.startOffset.orElse(pos.pointer).getOrElse[Int](0)
```
Comment (joan38) [inline bsp/src/mill/bsp/BspLoggedReporter.scala:97]: Not sure it might be `spacePointer` here. I need to do some testing.

**Comment by ckipp01** [comment]
Thanks for taking a look at this @joan38! Just tested it out, and much better!

Screenshot 2020-11-24 at 18 02 37

One other thing that I do notice though is that the diagnostic from Mill is as follows:
```
"diagnostics": [
    {
      "range": {
        "start": {
          "line": 8,
          "character": 15
        },
        "end": {
          "line": 8,
          "character": 15
        }
      },
      "severity": 1,
      "source": "compiler from mill",
      "message": "type mismatch;\n found   : String(\"hi\")\n required: Int"
    }
  ],
```
Whereas both with sbt and with Bloop it's:

```
"diagnostics": [
    {
      "range": {
        "start": {
          "line": 8,
          "character": 15
        },
        "end": {
          "line": 8,
          "character": 19
        }
      },
      "severity": 1,
      "source": "sbt",
      "message": "type mismatch;\n found   : String(\"hi\")\n required: Int"
    }
  ]
```

It still seems to work fine (I think there is some magic on the LSP client side) to maybe highlight the full symbol, but it'd be better to correctly capture the entire range and not return 15 for both start character and end character.

If it helps at all, here is the code sbt uses to get the range from the `Problem`.

https://github.com/sbt/sbt/blob/13b09bcd8fa0fde86aa4f4f8d7089fcb18e345c6/main/src/main/scala/sbt/internal/server/BuildServerReporter.scala#L133-L159

**Review by lefou — APPROVED** [review]
Looks good to me.

**Comment by lefou** [comment]
Merging this as it makes BSP error reporting significantly better. I'll open another issue for the open issue with the range end position.

**Comment by lefou** [comment]
Thanks you @joan38 for the fix and @ckipp01 for the feedback!

**Pull request merged**